pkg/service/one2onerestore/service_integration_test.go

```diff
@@ -57,67 +57,137 @@ func TestOne2OneRestoreServiceIntegration(t *testing.T) {
 		"location": loc,
 	})
 
-	Print("Truncate tables")
-	truncateAllTablesInKeyspace(t, clusterSession, ksName)
-	for _, tableName := range []string{BigTableName, mvName} {
-		if cnt := rowCount(t, clusterSession, ksName, tableName); cnt != 0 {
-			t.Fatalf("Unexpected row count: %d", cnt)
+	t.Run("Run 1-1-restore", func(t *testing.T) {
+		Print("Truncate tables")
+		truncateAllTablesInKeyspace(t, clusterSession, ksName)
+		for _, tableName := range []string{BigTableName, mvName} {
+			if cnt := rowCount(t, clusterSession, ksName, tableName); cnt != 0 {
+				t.Fatalf("Unexpected row count: %d", cnt)
+			}
+		}
+
+		Print("Run 1-1-restore")
+		h.runRestore(t, map[string]any{
+			"location":          loc,
+			"snapshot_tag":      tag,
+			"source_cluster_id": h.clusterID,
+			"nodes_mapping":     getNodeMappings(t, h.client),
+		})
+
+		Print("Validate data")
+		dstCnt := rowCount(t, clusterSession, ksName, BigTableName)
+		if srcCnt != dstCnt {
+			t.Fatalf("Expected row count in table %d, but got %d", srcCnt, dstCnt)
+		}
+		dstCntMV := rowCount(t, clusterSession, ksName, mvName)
+		if srcCntMV != dstCntMV {
+			t.Fatalf("Expected row count in materialized view %d, but got %d", srcCntMV, dstCntMV)
+		}
+		dstCntSI := rowCount(t, clusterSession, ksName, siTableName)
+		if srcCntSI != dstCntSI {
+			t.Fatalf("Expected row count in secondary index %d, but got %d", srcCntSI, dstCntSI)
 		}
-	}
 
-	Print("Run 1-1-restore")
-	h.runRestore(t, map[string]any{
-		"location":          loc,
-		"snapshot_tag":      tag,
-		"source_cluster_id": h.clusterID,
-		"nodes_mapping":     getNodeMappings(t, h.client),
+		// Ensure table's tombstone_gc mode is set to 'repair'
+		validateTombstoneGCMode(t, []testTable{
+			{
+				ks:           ksName,
+				name:         BigTableName,
+				expectedMode: modeRepair,
+			},
+			{
+				ks:           ksName,
+				name:         mvName,
+				isView:       true,
+				expectedMode: modeRepair,
+			},
+			// Until the https://github.com/scylladb/scylladb/issues/16454 is fixed,
+			// it is expected that the Secondary Index tombstone_gc mode will be 'timeout'.
+			{
+				ks:           ksName,
+				name:         siTableName,
+				isView:       true,
+				expectedMode: modeTimeout,
```

Comment:
nit: why is it expected? The current implementation treats indexes in the same way as materialized views (change tgc mode to repair, save schema, drop views, restore base tables, recreate views - with altered tgc mode). This means that this test will start failing for newer Scylla which fixes https://github.com/scylladb/scylladb/issues/16454. IMHO we shouldn't test that index tgc mode is set to timeout exactly, because we don't even care about it with the current workaround. I would not validate index tgc mode at all for now, and validate that it is set to repair starting from some Scylla release.